module/dice/build-roll.mjs

```diff
@@ -0,0 +1,17 @@
+/**
+ * Construct roll parts and populate its data object.
+ * @param {object} parts   Information on the parts to be constructed.
+ * @param {object} [data]  Roll data to use and populate while constructing the parts.
+ * @returns {{ parts: string[], data: object }}
+ */
+export default function buildRoll(parts, data={}) {
```

Comment:
Could be a static helper on `BasicRoll`. `buildRoll` feels not quite right since we're not returning a `Roll`. Since we're mutating the roll data, maybe something like `applyParts`?